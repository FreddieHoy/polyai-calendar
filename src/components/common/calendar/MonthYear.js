import React from 'react';
import styled from 'styled-components';
import { CalendarText } from 'config/FontStyles';

const Container = styled.div`
  border: solid 1px black;
  display: flex;
  justify-content: space-between;
`;

const Arrows = styled.p`
  margin: 0 10px 3px 10px;
  :hover {
    cursor: pointer;
  }
`;

const MonthYear = ({ year, month, switchMonth }) => (
  <Container>
    <Arrows onClick={switchMonth} id={''}>
      {'<'}
    </Arrows>
    <CalendarText>{`${month} ${year}`}</CalendarText>
    <Arrows onClick={switchMonth} id={'+'}>
      {'>'}
    </Arrows>
  </Container>
);

export default MonthYear;
