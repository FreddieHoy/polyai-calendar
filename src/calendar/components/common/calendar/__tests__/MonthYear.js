import { shallow } from 'enzyme';
import React from 'react';

import MonthYear from '../MonthYear';

describe('MonthYear', () => {
  it('renders all components correctly when given the correct props', () => {
    const props = {
      year: 2020,
      month: 'January',
      switchMonth: () => {}
    };
    const wrapper = shallow(<MonthYear {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('runs function passed as prop when clicked', () => {
    const props = {
      year: 2020,
      month: 'January',
      switchMonth: jest.fn()
    };
    const wrapper = shallow(<MonthYear {...props} />);
    const plusArrow = wrapper.find('#plus');
    plusArrow.simulate('click');

    expect(props.switchMonth).toBeCalledTimes(1);
  });
});
