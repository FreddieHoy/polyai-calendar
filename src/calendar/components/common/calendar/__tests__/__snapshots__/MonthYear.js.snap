// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthYear renders all components correctly when given the correct props 1`] = `
<styled.div>
  <styled.p
    id=""
    onClick={[Function]}
  >
    &lt;
  </styled.p>
  <styled.p>
    January 2020
  </styled.p>
  <styled.p
    id="plus"
    onClick={[Function]}
  >
    &gt;
  </styled.p>
</styled.div>
`;
